psi4/src/psi4/libmints/eribase.cc

```diff
@@ -2024,10 +2143,45 @@ void TwoElectronInt::init_shell_pairs12() {
                     sp->PB[i][j][1] = PB[1];
                     sp->PB[i][j][2] = PB[2];
                     sp->overlap[i][j] = pow(M_PI / gam, 3.0 / 2.0) * exp(-a1 * a2 * ab2 / gam) * c1 * c2;
+                    // Check overlap for screening condition
+                    if(fabs(sp->overlap[i][j]) >= overlap_cutoff) {
+                        // Make and populate fields of screened primitive pair
+                        PrimPairScreen pps;
+
+                        pps.P[0] = P[0];
+                        pps.P[1] = P[1];
+                        pps.P[2] = P[2];
+                        pps.PA[0] = PA[0];
+                        pps.PA[1] = PA[1];
+                        pps.PA[2] = PA[2];
+                        pps.PB[0] = PB[0];
+                        pps.PB[1] = PB[1];
+                        pps.PB[2] = PB[2];
+
+                        pps.ai = a1;
+                        pps.aj = a2;
+                        pps.gamma = gam;
+                        pps.ci = c1;
+                        pps.cj = c2;
+                        pps.overlap = sp->overlap[i][j];
+
+                        //Store this primitive pair in the shell pair
+                        spsc.nonzeroPrimPairs.push_back(pps);
```

Comment:
Probably should pre-allocate some space for the vector (you know what the maximum size is going to be).  `push_back` may perform multiple allocations and allocate too much.

Reply:
That's a good idea. I changed the `nonzeroPrimPairs` vector so that memory is reserved for `np_i*np_j` elements (the maximum number of primitive pairs). However, this still uses `push_back`. 

I'm a little hesitant to resize the vector to its maximum length, since  screening will ideally make the vector significantly shorter. Do you think that's okay, or should we use `resize`?

Reply:
You should be able to shrink it down to what it needs to be afterwards.

Reply:
That's perfect!
The `shrink_to_fit()` function reduces the capacity of the vector to match its size. Right now, I'm doing the following:

1. Make vector and resize to maximum size
2. Fill vector with primitive pairs via `[]`
3. Resize to final size (less than maximum because of screening)
4. Shrink to fit

Is this any better than doing the following, since `push_back()` is amortized constant time?

1. Make empty vector
2. Fill vector with primitive pairs via `push_back`
3. Shrink to fit